src/platform/utilities/oauth/constants.js

```diff
@@ -25,15 +25,30 @@ export const API_SIGN_IN_SERVICE_URL = ({
   `${environment.API_URL}/${version}/sign_in/${endpoint}${type &&
     `?type=${type}`}`;
 
-export const CLIENT_IDS = {
-  VAWEB: 'vaweb',
-  VAMOBILE: 'vamobile',
-  VAMOCK: 'vamock',
-  // TODO: refactor `CLIENT_IDS` to vary by environment. This is the value for
-  // the ARP frontend on localhost calling the ARP backend on localhost.
-  ARP: 'arp',
+const getClientIds = () => {
+  const baseClientIds = {
+    VAWEB: 'vaweb',
+    VAMOBILE: 'vamobile',
+    VAMOCK: 'vamock',
+  };
+
+  const envSpecificARPClientIds = (() => {
+    // NOTE: ARP prod sign-in (fe0d4b2cac7935e7eec5946b8ee31643) is not enabled in prod yet and a formal request must be made. See:
+    // https://github.com/department-of-veterans-affairs/vets-website/pull/30878#discussion_r1681519493
+    if (environment.isProduction()) return 'fe0d4b2cac7935e7eec5946b8ee31643'; // SOURCE: https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/blob/09e2f044cfe4d67a9f2c7ce22606facd18d381ca/apps/vets-api/prod/values.yaml#L2367
+    if (environment.isStaging()) return 'ce6db4d7974daf061dccdd21ba9add14'; // SOURCE: https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/blob/09e2f044cfe4d67a9f2c7ce22606facd18d381ca/apps/vets-api/staging/values.yaml#L2362
+    if (environment.isDev()) return 'arp'; // NOTE: Same as localhost: https://github.com/department-of-veterans-affairs/vets-website/pull/30878#issuecomment-2233940395
+    return 'arp';
```

Comment:
Seems like these lines could be simplified to just `return 'arp'`?

Reply:
Yeah, I was thinking it would be good to explicitly call out `if (environment.isDev()) return 'arp';` so it's clear that `isDev` was considered in this implementation given confusion around that environment (see [comment 1](https://github.com/department-of-veterans-affairs/vets-website/pull/30878#issuecomment-2233876507) and [comment 2](https://github.com/department-of-veterans-affairs/vets-website/pull/30878#issuecomment-2233940395)). I could just remove `if (environment.isDev()) return 'arp';` and make a clarifying comment.